File: src/main/java/com/ziwok/airticketsystem/api/model/Flight.java
package com.ziwok.airticketsystem.api.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Getter
@Setter
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "FLIGHTS")
public class Flight {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer flightId;

    @Column(unique = true, nullable = false)
    private String flightNumber;

    private LocalDateTime departureTime;

    private LocalDateTime arrivalTime;

    private String origin;

    private String destination;

    @ManyToOne
    @JoinColumn(name = "aircraft_id", nullable = false)
    private Aircraft aircraft;
}

File: internal/entity/flight.go
package entity

import (
	"time"
)

// Flight represents a flight record.
type Flight struct {
	FlightID           int       `json:"flight_id"`
	AircraftID         int       `json:"aircraft_id"`
	DepartureAirport   string    `json:"departure_airport"`
	DestinationAirport string    `json:"destination_airport"`
	DepartureTime      time.Time `json:"departure_time"`
	DestinationTime    time.Time `json:"destination_time"`
	Status             string    `json:"status"`
	CreatedAt          time.Time `json:"created_at"`
	UpdatedAt          time.Time `json:"updated_at"`
}
